core/src/com/punchbrain/awake/model/Circuit.java

```diff
@@ -23,28 +29,57 @@ public Circuit(GameObject lamp, GameObject flipSwitch, Light lightObject, AudioM
         this.audioManager = audioManager;
     }
 
-    public boolean isOn(){
-        return isOn;
+
+    public Circuit updatePassiveBehaviour(float delta) {
```

Comment:
What do you think about this? It puts the behaviour on the model, so that the behaviour class is more like a controller.